server/controllers/message.js

```diff
@@ -0,0 +1,113 @@
+const Message = require("../models/Message");
+const asyncHandler = require("express-async-handler");
+const Conversation = require("../models/Conversation");
+
+exports.getAllMessages = asyncHandler(async (req, res, next) => {
+  const messages = await Message.find({
+    conversation_id: req.params.conversationId,
+  });
+
+  if (!messages) {
+    res.status(404);
+    throw new Error("No messages found for given id");
+  }
+
+  res.status(200).json({ messages: messages });
+});
+
+exports.createMessage = asyncHandler(async (req, res, next) => {
+  const { conversation_id, body } = req.body;
+
+  console.log(req.user);
+  let conversation = await Conversation.findById(conversation_id);
+
+  if (conversation) {
+    if(conversation.participants.includes(req.user.id)){
+      const createdMessage = await Message.create({
+        conversation_id,
+        body,
+        creator: req.user.id,
+      });
+  
+      if (createdMessage) {
+        let messageArr = conversation.messages;
+        messageArr.push(createdMessage._id);
```

Comment:
```suggestion
        const messageArr = conversation.messages;
        messageArr.push(createdMessage._id);
```
You can push to a constant array, you just can't reinitialize it